modules/common/monitoring.tf

```diff
@@ -0,0 +1,55 @@
+resource "kubernetes_namespace" "monitoring" {
+  count = var.monitoring_enabled == true ? 1 : 0
+  metadata {
+    name = "kube-monitoring"
+  }
+}
+
+resource "helm_release" "prometheus_monitoring_stack" {
+  count      = var.monitoring_enabled == true ? 1 : 0
+  depends_on = [module.eks, module.vpc]
+  name       = "prometheus"
+  repository = "https://prometheus-community.github.io/helm-charts"
+  chart      = "kube-prometheus-stack"
+  version    = "45.21.0"
+  timeout    = 600 # it takes a while to install this Helm chart
+  namespace  = kubernetes_namespace.monitoring[count.index].metadata[0].name
+
+  values = [
+    yamlencode({
+      grafana = {
+        service = {
+          type                     = var.monitoring_grafana_expose_lb == true ? "LoadBalancer" : "ClusterIP",
+          loadBalancerSourceRanges = var.whitelist_cidr
+        },
+        persistence = {
+          enabled = true
+          size    = "10Gi"
+        },
+        sidecar = {
+          dashboards = {
+            enabled    = true,
+            label      = "grafana_dashboard",
+            labelValue = "dc_monitoring"
+          }
+        }
+      },
+      prometheus = {
+        prometheusSpec = {
+          storageSpec = {
+            volumeClaimTemplate = {
+              spec = {
+                accessModes = ["ReadWriteOnce"],
+                resources = {
+                  requests = {
+                    storage = "10Gi"
```

Comment:
10Gi should be enough for short running instances I assume? Curious to know the growth rate of data with current configuration.

Reply:
Good point. Yes, potentially more storage can be required, though it depends on the number of metrics, as well as metrics cardinality and Prometheus retention period.

I have added 2 more variables:

```
prometheus_pvc_disk_size
grafana_pvc_disk_size
```

as well as an info box/comments saying that you can't resize the volume once created, so if you believe your cluster will be busy, override the pvc storage claim when deploying monitoring stack for the first time.